notes/stv/20200528-tmp-dir-issue.txt

```diff
@@ -0,0 +1,461 @@
+spark.driver.maxResultSize	8192m
+spark.local.dir         /tmp/spark-temp
+spark.executor.cores            5
+#spark.dynamicAllocation.enabled         true
+#spark.shuffle.service.enabled           true
+
+
+# Looks like this is set here: spark.local.dir   /tmp/spark-temp
+# /tmp/ directories are probably getting cleared, and running Spark notebooks can't find them anymore
+
+
+## Check Fedora /tmp/ rules:
+## ----------------------------------
+ 
+cat /usr/lib/tmpfiles.d/tmp.conf 
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+# See tmpfiles.d(5) for details
+
+# Clear tmp directories separately, to make them easier to override
+q /tmp 1777 root root 10d
+q /var/tmp 1777 root root 30d
+
+# Exclude namespace mountpoints created with PrivateTmp=yes
+x /tmp/systemd-private-%b-*
+X /tmp/systemd-private-%b-*/tmp
+x /var/tmp/systemd-private-%b-*
+X /var/tmp/systemd-private-%b-*/tmp
+
+# Remove top-level private temporary directories on each boot
+R! /tmp/systemd-private-*
+R! /var/tmp/systemd-private-*
+
+
+## Change spark local dir parameter
+## ----------------------------------
+nano /home/fedora/spark/conf/spark-defaults.conf
+..
+
+spark.master                     yarn
+spark.driver.memory              17g
+spark.yarn.am.memory            17g
+spark.executor.memory          17g
+spark.eventLog.enabled  true
+spark.eventLog.dir hdfs://stv-dev-master:9000/spark-log
+spark.driver.maxResultSize	8192m
+spark.local.dir         /home/fedora/spark/local
+spark.executor.cores            5
+#spark.dynamicAllocation.enabled         true
+#spark.shuffle.service.enabled           true
+
+..
+
+
+## Restart Zeppelin
+## ----------------------------------
+
+/home/fedora/zeppelin/bin/zeppelin-daemon.sh stop
+/home/fedora/zeppelin/bin/zeppelin-daemon.sh start
+
+
+
+ls -al /home/fedora/spark/local/
+total 16
+drwxr-xr-x.  4 fedora fedora 4096 May 28 17:27 .
+drwxrwxr-x. 14 fedora fedora 4096 May 28 17:07 ..
+drwxrwxr-x.  5 fedora fedora 4096 May 28 17:30 blockmgr-7a28d729-2842-4b16-803b-a6082d30a0fd
+drwx------.  4 fedora fedora 4096 May 28 17:27 spark-a55fe940-ba73-4b62-a30c-4e5df79d0ef7
+
+
+
+## Try running again:
+## ----------------------
+
+# Try running example again
+# In Zeppelin: 
+
+%spark.pyspark
+
+import random
+NUM_SAMPLES = 2000000000
+
+def inside(p):
+    x, y = random.random(), random.random()
+    return x*x + y*y < 1
+
+count = sc.parallelize(xrange(0, NUM_SAMPLES)) \
+             .filter(inside).count()
+print "Pi is roughly %f" % (4.0 * count / NUM_SAMPLES)
+
+Pi is roughly 3.141510
+
+
+
+# All other Notebooks seem to be running successfully now.
+#
+# We need to now:
+#
+#   1) Check whether this happens again with the existing configuration
```

Comment:
Can we create an issue to track these.

Reply:
Actually, looks like you already did this.
https://github.com/wfau/aglais/issues/112